kcidb/monitor/spool/__init__.py

```diff
@@ -97,29 +98,49 @@ def post(self, notification, timestamp=None):
             timestamp:      An "aware" datetime.datetime object specifying the
                             notification creation time, or None to use
                             datetime.datetime.now(datetime.timezone.utc).
+            due:            An "aware" datetime.datetime object specifying the
+                            time notification should be sent out,
+                            or None to use "timestamp".
 
         Returns:
-            True, if the notification was posted onto the spool,
+            True, if the notification was posted onto the spool
+            or an existing notification with timestamp past the
+            picked_until time was updated,
             False, if not (it was already there).
         """
         assert isinstance(notification, Notification)
         assert timestamp is None or \
-            isinstance(timestamp, datetime.datetime) and timestamp.tzinfo
+               isinstance(timestamp, datetime.datetime) and timestamp.tzinfo
         if timestamp is None:
             timestamp = datetime.datetime.now(datetime.timezone.utc)
+        assert due is None or \
+               isinstance(due, datetime.datetime) and due.tzinfo
+        if due is None:
+            due = timestamp
+        message = notification.render().as_string(policy=email.policy.SMTPUTF8)
 
-        try:
-            self._get_doc(notification.id).create(dict(
-                created_at=timestamp,
-                # Set to a definitely timed-out time, free for picking
-                picked_until=datetime.datetime.min,
-                message=notification.render().as_string(
-                    policy=email.policy.SMTPUTF8
-                ),
-            ))
-        except Conflict:
-            return False
-        return True
+        @firestore.transactional
+        def create_or_update(transaction, notification, message):
+            snapshot = self._get_doc(notification.id). \
+                get(field_paths=["picked_until"], transaction=transaction)
+            if timestamp > snapshot.get("picked_until"):
+                transaction. \
+                    update(self._get_doc(notification.id), dict(
+                           created_at=timestamp,
+                           message=message))
```

Comment:
In general, could you please wrap after the opening parenthesis, and keep each argument on its own line(s), if wrapping? E.g. like this:
```suggestion
                transaction.update(
                    self._get_doc(notification.id),
                    dict(created_at=timestamp, message=message)
                )
```
But read below, as I think we don't need this call.

Reply:
Also, we shouldn't be updating the `created_at` timestamp, since we're just updating the notification, not creating a new one.